pkg/rpc/context.go

```diff
@@ -1111,9 +1111,11 @@ func (d *delayingConn) Read(b []byte) (n int, err error) {
 			return 0, errors.WithStack(errMagicNotFound)
 		}
 
-		// Once we receive our first packet, we set our delay to the expected
-		// delay that was sent on the write side.
-		d.latency = time.Duration(hdr.DelayMS) * time.Millisecond
+		// Once we receive our first packet with a DelayMS field set, we set our
+		// delay to the expected delay that was sent on the write side.
```

Comment:
I think it would be nice to augment this comment with some more explanation about what this conditional is about, considering how confused we were trying to understand what was going on. I'm thinking about an explanation of when this check actually gets hit, and how the sender is the one responsible for setting the latency, and so on.

Reply:
I added something but TBH, I don't fully understand all of the reasons why this was broken just yet.  The comment explains that, as well as the rationale for why it's safe to not overwrite a non-zero value.  Let me know if this is along the lines of what you were hoping to see.